### PR TITLE
fix(GH-3684): aggregate CloudIntegrationEvents for connector task executions

### PR DESCRIPTION
**Pull request by igdianov**
This PR improves handling of CloudIntegrationEvents to ensure they are aggregated inside engine transaction context to be able to populate execution context headers in the audit message after transaction closes. The following events have been refactored:

- [x] CloudIntegrationRequestedEvent
- [x] CloudIntegrationResultReceivedEvent
- [x] CloudIntegrationErrorReceivedEvent

Part of https://github.com/Activiti/Activiti/issues/3684

### CODE REVIEW

File: activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/CompositeCommand.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package org.activiti.services.connectors.channel;
+
+import org.activiti.engine.impl.interceptor.Command;
+import org.activiti.engine.impl.interceptor.CommandContext;
+
+import java.util.ArrayList;
+
+class CompositeCommand extends ArrayList<Command<?>> implements Command<Void> {
```
Comment (erdemedeiros) [inline activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/CompositeCommand.java]: I'd prefer not to have `CompositeCommand` extending `ArrayList`, I'd rather receive a list of commands in the constructor and maybe add a utility method to initialize it:
```java
CompositeCommand.of(new Command1(), new Command2())
```

File: activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/ServiceTaskIntegrationErrorEventHandler.java [hunk 2]
```diff
@@ -94,17 +83,10 @@ public void receive(IntegrationError integrationError) {
...
-                        CloudBpmnError cloudBpmnError = new CloudBpmnError(errorCode,
-                                                                           integrationError.getErrorMessage());
-                        cloudBpmnError.setStackTrace(integrationError.getStackTraceElements()
-                                                                     .toArray(new StackTraceElement[] {}));
                         try {
-                            managementService.executeCommand(new PropagateCloudBpmnErrorCmd(cloudBpmnError,
-                                                                                            execution));
+                            managementService.executeCommand(new TriggerIntegrationContextErrorCmd(integrationError,
+                                                                                                   execution));
+                            return;
```
Comment (erdemedeiros) [inline activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/ServiceTaskIntegrationErrorEventHandler.java:83]: I was wondering if we can avoid this `return`.

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
I have some concerns about `CompositeCommand`. LGTM otherwise.

**Review by erdemedeiros — APPROVED** [review]

File: activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/CompositeCommand.java (hunk 1 above)
Comment (igdianov) [reply activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/CompositeCommand.java]: fixed

Comment (igdianov) [reply activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-connectors/src/main/java/org/activiti/services/connectors/channel/CompositeCommand.java]: sounds good

**Pull request merged**